c_src/notif_set.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+#include <stdio.h>
+
+/* forward declarations */
+struct Notif_set;
+struct Notif_set_itr;
+struct Message;
+
+struct Notif_set *notif_set_new();
+int notif_set_destroy(struct Notif_set *tab);
+int notif_set_insert(struct Notif_set *tab, struct Message *msg);
```

Comment:
When passing in pointer params, consider making them `const` if you the function doesn't change them. In this case, I originally was concerned that you passed a temporary object to a function that was planning on modifying it.